orchestrator/db/sorting/process.py

```diff
@@ -17,22 +17,29 @@
 }
 
 
-def generic_process_relation_sort(field: Column) -> Callable[[QueryType, SortOrder], QueryType]:
+def generic_process_relation_sort(
+    field: Column, join_tables: list | None = None
+) -> Callable[[QueryType, SortOrder], QueryType]:
+    join_tables = join_tables or []
+
     def sort_function(query: QueryType, order: SortOrder) -> QueryType:
         if isinstance(query, Select):
-            joined_query: QueryType = query.join(ProcessSubscriptionTable).join(SubscriptionTable).join(ProductTable)
-        else:
-            # Can't join on CompoundSelect
-            joined_query = query
-        if order == SortOrder.DESC:
-            return joined_query.order_by(expression.desc(field))
-        return joined_query.order_by(expression.asc(field))
+            joined_tables = [str(t[0]) for t in query._setup_joins]
+
+            for table in join_tables:
+                if (tablename := getattr(table, "__tablename__", "")) and tablename not in joined_tables:
+                    query = query.join(table)
```

Comment:
What's so bad about multiple joins that we hook into 2 sqlalchemy internal variables to prevent it?

Reply:
If you repeat the same join, sqlalchemy will get confused. That can happen when we use a composite sort order, like

`order_by=[{"field":"productTag", "order":"ASC"},{"field":"productName","value":"firewall"}]`

Maybe it can be solved a different way that I overlooked, like join aliases.

Reply:
I think ideally these sort functions should return the order_by expression and additionally the table(s) that it needs to join, and that a higher level function gathers the unique tables from all sort functions and creates the joins. But that might be a pretty big change :) 

Scanning the sqlalchemy docs/code I don't suspect join aliases will solve this but maybe I'm overlooking something. Anyway, I'm also fine with a small comment in the code explaining the necessity, and possibly outlining a better solution as mentioned above - if you think it's plausible.

Reply:
That would be a quite bigger change. I'm fine with a small comment for now

Reply:
Issue created #656